Lib/argparse.py

```diff
@@ -2553,11 +2557,30 @@ def _get_value(self, action, arg_string):
         return result
 
     def _check_value(self, action, value):
+        if not action.choices and isinstance(action.choices, list):
+            msg = 'Either add options in choices array or remove it'
+            raise ArgumentError(action, msg)
+
         # converted value must be one of the choices (if specified)
         if action.choices is not None and value not in action.choices:
-            args = {'value': value,
-                    'choices': ', '.join(map(repr, action.choices))}
+            args = {
+                'value': value,
+                'choices': ', '.join(map(repr, action.choices)),
+            }
             msg = _('invalid choice: %(value)r (choose from %(choices)s)')
+
+            if self.suggest_on_error:
+                try:
+                    import difflib
+                    closest_choice = difflib.get_close_matches(value, action.choices, 1)
+                    if closest_choice:
+                        closest_choice = closest_choice[0]
+                        args['closest'] = closest_choice
```

Comment:
I would use `suggestion` instead of `closest`.

Reply:
I meant the key. The translated string is the part of the API, users will use it in their translations, so we have to think carefully about which names to use.